atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -78,79 +47,52 @@ def run(self):
-        # just use oc image extract to get image content
-#        container_dict = self.tasker.create_container(image, command=['/bin/bash'])
-#        container_id = container_dict['Id']
-#        try:
-#            bits, _ = self.tasker.get_archive(container_id,
-#                                              IMG_MANIFESTS_PATH)
-#        except APIError as ex:
-#            msg = ('Could not extract operator manifest files. '
-#                   'Is there a %s path in the image?' % (IMG_MANIFESTS_PATH))
-#            self.log.debug('Error while trying to extract %s from image: %s',
-#                           IMG_MANIFESTS_PATH, ex)
-#            self.log.error(msg)
-#            raise RuntimeError('%s %s' % (msg, ex)) from ex
-#
-#        except Exception as ex:
-#            raise RuntimeError('%s' % ex) from ex
-#
-#        finally:
-#            try:
-#                self.tasker.remove_container(container_id)
-#            except Exception as ex:
-#                self.log.warning('Failed to remove container %s: %s', container_id, ex)
-        bits = b''
-
-        with tempfile.NamedTemporaryFile() as extracted_file:
-            for chunk in bits:
-                extracted_file.write(chunk)
-            extracted_file.flush()
-            tar_archive = tarfile.TarFile(extracted_file.name)
-
-        tar_archive.extractall(manifests_archive_dir)
-        manifests_path = os.path.join(manifests_archive_dir, MANIFESTS_DIR_NAME)
+        platforms = get_platforms(self.workflow.data)
+        image = self.workflow.data.tag_conf.get_unique_images_with_platform(platforms[0])[0]
+        tmp_dir = tempfile.mkdtemp(dir=self.workflow.build_dir.platform_dir(platforms[0]).path)
+        manifests_dir = os.path.join(tmp_dir, MANIFESTS_DIR_NAME)
+
+        self.workflow.imageutil.extract_file_from_image(image, IMG_MANIFESTS_PATH, tmp_dir)
 
         if has_operator_bundle_manifest(self.workflow):
-            self._verify_csv(manifests_path)
+            self._verify_csv(manifests_dir)
 
-        manifests_zipfile_path = os.path.join(manifests_archive_dir, OPERATOR_MANIFESTS_ARCHIVE)
+        manifests_zipfile_path = (self.workflow.build_dir.any_platform.path /
+                                  OPERATOR_MANIFESTS_ARCHIVE)
         with zipfile.ZipFile(manifests_zipfile_path, 'w') as archive:
-            for root, _, files in os.walk(manifests_path):
+            for root, _, files in os.walk(manifests_dir):
                 for f in files:
-                    filedir = os.path.relpath(root, manifests_path)
+                    filedir = os.path.relpath(root, manifests_dir)
                     filepath = os.path.join(filedir, f)
                     archive.write(os.path.join(root, f), filepath, zipfile.ZIP_DEFLATED)
             manifest_files = archive.namelist()
-            if not manifest_files:
-                self.log.error('Empty operator manifests directory')
-                raise RuntimeError('Empty operator manifests directory')
```

Comment:
Why remove this check?

Edit: is it because extract_file_from_image already sort of does that?